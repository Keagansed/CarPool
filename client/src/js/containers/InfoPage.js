// File Type: Component

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import AppBar from '@material-ui/core/AppBar';
import Button from '@material-ui/core/Button';
import CarIcon from '@material-ui/icons/DirectionsCar';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import CssBaseline from '@material-ui/core/CssBaseline';
import Grid from '@material-ui/core/Grid';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import { withStyles } from '@material-ui/core/styles';
import Avatar from '@material-ui/core/Avatar';
import Dialog from '@material-ui/core/Dialog';
import { DialogContent, DialogContentText } from '@material-ui/core';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogActions from '@material-ui/core/DialogActions';

import logo from "../../css/images/logo.png";
import communityImage from "../../css/images/CommunityImage.jpeg";
import moneyImage from "../../css/images/MoneyImage.jpeg";
import commuteImage from "../../css/images/CommuteImage.jpeg";
import roadImage from "../../css/images/RoadImage.jpeg";
import ios1 from "../../css/images/ios1.JPG";
import ios2 from "../../css/images/ios2.JPG";
import android1 from "../../css/images/android1.jpg";
import android2 from "../../css/images/android2.jpg";

const styles = theme => ({
    appBar: {
        position: 'relative',
    },
    icon: {
        marginRight: theme.spacing.unit * 2,
    },
    heroUnit: {
        backgroundColor: theme.palette.background.paper,
    },
    heroContent: {
        maxWidth: 600,
        margin: '0 auto',
        padding: `${theme.spacing.unit * 4}px 0 ${theme.spacing.unit * 4}px`,
    },
    heroButtons: {
        marginTop: theme.spacing.unit * 4,
    },
    layout: {
        width: 'auto',
        marginLeft: theme.spacing.unit * 3,
        marginRight: theme.spacing.unit * 3,
        [theme.breakpoints.up(1100 + theme.spacing.unit * 3 * 2)]: {
            width: 1100,
            marginLeft: 'auto',
            marginRight: 'auto',
        },
    },
    cardGrid: {
        padding: `${theme.spacing.unit * 4}px 0`,
    },
    card: {
        height: '100%',
        display: 'flex',
        flexDirection: 'column',
    },
    cardMedia: {
        paddingTop: '56.25%', // 16:9
    },
    cardContent: {
        flexGrow: 1,
    },
    footer: {
        backgroundColor: theme.palette.background.paper,
        padding: theme.spacing.unit * 6,
    },
    carpoolLogo: {
        width: 252,
        height: 252,
        marginLeft: 'auto',
        marginRight: 'auto',
        marginBottom: theme.spacing.unit * 1,
    },
    dialogImg: {
        width: "100%",
        marginTop: 7,
        marginBottom: 7,
        border: "white solid 1px",
        justify: "center"
    }
});

/*
* Purpose: Landing page compenent is the first page the user sees when opening the app
*/
class InfoPage extends Component {
    constructor() {
        super();

        this.state = {
            openDialog: false,
        }
    }

    closeDialog = () => {
        this.setState({
            openDialog: false,
        })
    }

    openDialog = () => {
        this.setState({
            openDialog: true,
        })
    }
    
    renderDownloadDialog = () => {
        const { classes } = this.props;

        return (
            <Dialog
                open={this.state.openDialog}
                onClose={this.closeDialog}
                maxWidth="lg"   
                scroll='paper' 
            >
                <DialogTitle>Download App</DialogTitle>
                <DialogContent style={{ justifyContent: "center" }}>
                    <DialogContentText variant="title">
                        Android
                    </DialogContentText>

                    <img src={android1} alt="" className={classes.dialogImg}/>
                    <img src={android2} alt="" className={classes.dialogImg}/>

                    <DialogContentText variant="title">
                        iOS
                    </DialogContentText>

                    <img src={ios1} alt="" className={classes.dialogImg}/>
                    <img src={ios2} alt="" className={classes.dialogImg}/>

                </DialogContent>
                <DialogActions>
                    <Button onClick={this.closeDialog} color="primary">
                        Close
                    </Button>
                </DialogActions>
            </Dialog>
        )
    }

    render() {
        const { classes } = this.props;
        return (
            <React.Fragment>
                {this.renderDownloadDialog()}
                <CssBaseline />
                <AppBar position="static" className={classes.appBar}>
                    <Toolbar>
                        <CarIcon className={classes.icon} />
                        <Typography variant="title" color="inherit" noWrap>
                            The Iminsys Carpool Platform
                        </Typography>
                    </Toolbar>
                </AppBar>
                <main>
                    {/* Hero unit */}
                    <div className={classes.heroUnit}>
                        <div className={classes.heroContent}>
                            <Avatar src={logo} align='center' className={classes.carpoolLogo} />
                            <Typography variant="title" align="center" color="textSecondary">
                                Welcome to the ridesharing platform that helps you help eachother.
                            </Typography>
                            <div className={classes.heroButtons}>
                                <Grid container spacing={16} justify="center">
                                    <Grid item>
                                        <Button href={`/Landing`} variant="contained" color="primary">
                                            Get Started
                                        </Button>
                                    </Grid>
                                    <Grid item>
                                        <Button variant="outlined" color="primary" disabled>
                                            Watch Video
                                        </Button>
                                    </Grid>
                                </Grid>
                                <Grid container spacing={16} justify="center" style={{ marginTop: 16, marginBottom: -35 }} onClick={this.openDialog}>
                                    <Grid item>
                                        <Button variant="contained" color="primary">
                                            Download
                                        </Button>
                                    </Grid>
                                </Grid>
                            </div>
                        </div>
                    </div>
                    <div className={classNames(classes.layout, classes.cardGrid)}>
                        {/* End hero unit */}
                        <Grid container spacing={40}>
                            <Grid item sm={6} md={4} lg={3}>
                                <Card className={classes.card}>
                                    <CardMedia
                                        className={classes.cardMedia}
                                        image={communityImage}
                                        title="Community"
                                    />
                                    <CardContent className={classes.cardContent}>
                                        <Typography gutterBottom variant="headline" component="h2">
                                            Join a Community
                                        </Typography>
                                        <Typography>
                                            The intention of the Iminsys Carpool Platform is to bring commuters together and to
                                            help them help eachother. By giving users the tools to find others who travel along
                                            similar routes, interact with them, and share driving responsibilities with no costs involved,
                                            the platform exposes the good in people and creates a network out of which everyone benefits.
                                        </Typography>
                                    </CardContent>
                                </Card>
                            </Grid>
                            <Grid item sm={6} md={4} lg={3}>
                                <Card className={classes.card}>
                                    <CardMedia
                                        className={classes.cardMedia}
                                        image={moneyImage}
                                        title="Money"
                                    />
                                    <CardContent className={classes.cardContent}>
                                        <Typography gutterBottom variant="headline" component="h2">
                                            Save on Travel
                                        </Typography>
                                        <Typography>
                                            Unfortunately, travel is expensive. This is something that we are all familiar with
                                            and it is not a problem that is going away anytime soon. Our Carpool Platform does
                                            not provide for the exchange of money; however, it does allow people to alternate driving
                                            responsibilities so the costs involved in driving are reduced for everyone.
                                        </Typography>
                                    </CardContent>
                                </Card>
                            </Grid>
                            <Grid item sm={6} md={4} lg={3}>
                                <Card className={classes.card}>
                                    <CardMedia
                                        className={classes.cardMedia}
                                        image={commuteImage}
                                        title="Commute"
                                    />
                                    <CardContent className={classes.cardContent}>
                                        <Typography gutterBottom variant="headline" component="h2">
                                            Enjoy Commuting
                                        </Typography>
                                        <Typography>
                                            Driving the same route on a daily basis is a tedious task but when you're driving with new
                                            friends it will become something you look forward to. Our Carpool
                                            Platform makes numerous provisions to ensure you only travel with people you feel comfortable
                                            around, so the only question that remains is - why drive alone when you can drive together?
                                        </Typography>
                                    </CardContent>
                                </Card>
                            </Grid>
                            <Grid item sm={6} md={4} lg={3}>
                                <Card className={classes.card}>
                                    <CardMedia
                                        className={classes.cardMedia}
                                        image={roadImage}
                                        title="Image title"
                                    />
                                    <CardContent className={classes.cardContent}>
                                        <Typography gutterBottom variant="headline" component="h2">
                                            Clean our Roads
                                        </Typography>
                                        <Typography>
                                            If you've ever had to sit in peak hour traffic you'll know what a pain it can be. A big reason 
                                            for unnecessary traffic is that there are too many cars with one driver and
                                            a host of empty seats. By driving together, we are reducing the number cars on our roads, not only
                                            minimizing traffic, but minimizing the environmental impact our vehicles have.
                                        </Typography>
                                    </CardContent>
                                </Card>
                            </Grid>
                        </Grid>
                    </div>
                </main>
                {/* Footer */}
                <footer className={classes.footer}>
                    <Typography variant="title" align="center" gutterBottom>
                        Created By
                    </Typography>
                    <Typography variant="subheading" align="center" color="textSecondary" component="p">
                        The Brogrammers Development Team
                    </Typography>
                </footer>
                {/* End footer */}
            </React.Fragment>
        );
    }
}

InfoPage.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(InfoPage);