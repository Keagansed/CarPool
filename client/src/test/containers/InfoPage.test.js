import React from 'react';
import InfoPage from '../../js/containers/InfoPage';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

describe('InfoPage Component', () => {

    it ('captures snapshot', () => {
        const renderedValue = renderer.create(<MemoryRouter><InfoPage /></MemoryRouter>).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

    it ('renders correctly', () => {
        let wrapper = shallow(<MemoryRouter><InfoPage /></MemoryRouter>)
        expect(wrapper.length).toEqual(1);
    });
});