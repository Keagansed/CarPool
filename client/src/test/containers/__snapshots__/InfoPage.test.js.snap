// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPage Component captures snapshot 1`] = `
Array [
  <header
    className="MuiPaper-root-22 MuiPaper-elevation4-28 MuiAppBar-root-13 MuiAppBar-positionStatic-17 MuiAppBar-colorPrimary-20 InfoPage-appBar-1"
  >
    <div
      className="MuiToolbar-root-49 MuiToolbar-regular-51 MuiToolbar-gutters-50"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-53 InfoPage-icon-2"
        color={undefined}
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5h-11c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16zm11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zM5 11l1.5-4.5h11L19 11H5z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
      <h2
        className="MuiTypography-root-62 MuiTypography-title-68 MuiTypography-colorInherit-82 MuiTypography-noWrap-79"
      >
        The Iminsys Carpool Platform
      </h2>
    </div>
  </header>,
  <main>
    <div
      className="InfoPage-heroUnit-3"
    >
      <div
        className="InfoPage-heroContent-4"
      >
        <div
          align="center"
          className="MuiAvatar-root-88 InfoPage-carpoolLogo-12"
        >
          <img
            alt={undefined}
            className="MuiAvatar-img-90"
            sizes={undefined}
            src="logo.png"
            srcSet={undefined}
          />
        </div>
        <h2
          className="MuiTypography-root-62 MuiTypography-title-68 MuiTypography-colorTextSecondary-86 MuiTypography-alignCenter-76"
        >
          Welcome to the ridesharing platform that helps you help eachother.
        </h2>
        <div
          className="InfoPage-heroButtons-5"
        >
          <div
            className="MuiGrid-container-91 MuiGrid-spacing-xs-16-114 MuiGrid-justify-xs-center-108"
          >
            <div
              className="MuiGrid-item-92"
            >
              <a
                className="MuiButtonBase-root-214 MuiButton-root-188 MuiButton-contained-199 MuiButton-containedPrimary-200 MuiButton-raised-202 MuiButton-raisedPrimary-203"
                href="/Landing"
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                <span
                  className="MuiButton-label-189"
                >
                  Get Started
                </span>
                <span
                  className="MuiTouchRipple-root-217"
                />
              </a>
            </div>
            <div
              className="MuiGrid-item-92"
            >
              <button
                className="MuiButtonBase-root-214 MuiButtonBase-disabled-215 MuiButton-root-188 MuiButton-text-190 MuiButton-textPrimary-191 MuiButton-outlined-196 MuiButton-outlinedPrimary-197 MuiButton-disabled-208"
                disabled={true}
                onBlur={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex="-1"
                type="button"
              >
                <span
                  className="MuiButton-label-189"
                >
                  Watch Video
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="InfoPage-layout-6 InfoPage-cardGrid-7"
    >
      <div
        className="MuiGrid-container-91 MuiGrid-spacing-xs-40-117"
      >
        <div
          className="MuiGrid-item-92 MuiGrid-grid-sm-6-139 MuiGrid-grid-md-4-151 MuiGrid-grid-lg-3-164"
        >
          <div
            className="MuiPaper-root-22 MuiPaper-elevation1-25 MuiPaper-rounded-23 MuiCard-root-224 InfoPage-card-8"
          >
            <div
              className="MuiCardMedia-root-225 InfoPage-cardMedia-9"
              src={undefined}
              style={
                Object {
                  "backgroundImage": "url(\\"CommunityImage.jpeg\\")",
                }
              }
              title="Community"
            />
            <div
              className="MuiCardContent-root-227 InfoPage-cardContent-10"
            >
              <h2
                className="MuiTypography-root-62 MuiTypography-headline-67 MuiTypography-gutterBottom-80"
              >
                Join a Community
              </h2>
              <p
                className="MuiTypography-root-62 MuiTypography-body1-71"
              >
                The intention of the Iminsys Carpool Platform is to bring commuters together and to help them help eachother. By giving users the tools to find others who travel along similar routes, interact with them, and share driving responsibilities with no costs involved, the platform exposes the good in people and creates a network out of which everyone benefits.
              </p>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-92 MuiGrid-grid-sm-6-139 MuiGrid-grid-md-4-151 MuiGrid-grid-lg-3-164"
        >
          <div
            className="MuiPaper-root-22 MuiPaper-elevation1-25 MuiPaper-rounded-23 MuiCard-root-224 InfoPage-card-8"
          >
            <div
              className="MuiCardMedia-root-225 InfoPage-cardMedia-9"
              src={undefined}
              style={
                Object {
                  "backgroundImage": "url(\\"MoneyImage.jpeg\\")",
                }
              }
              title="Money"
            />
            <div
              className="MuiCardContent-root-227 InfoPage-cardContent-10"
            >
              <h2
                className="MuiTypography-root-62 MuiTypography-headline-67 MuiTypography-gutterBottom-80"
              >
                Save on Travel
              </h2>
              <p
                className="MuiTypography-root-62 MuiTypography-body1-71"
              >
                Unfortunately, travel is expensive. This is something that we are all familiar with and it is not a problem that is going away anytime soon. Our Carpool Platform does not provide for the exchange of money; however, it does allow people to alternate driving responsibilities so the costs involved in driving are reduced for everyone.
              </p>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-92 MuiGrid-grid-sm-6-139 MuiGrid-grid-md-4-151 MuiGrid-grid-lg-3-164"
        >
          <div
            className="MuiPaper-root-22 MuiPaper-elevation1-25 MuiPaper-rounded-23 MuiCard-root-224 InfoPage-card-8"
          >
            <div
              className="MuiCardMedia-root-225 InfoPage-cardMedia-9"
              src={undefined}
              style={
                Object {
                  "backgroundImage": "url(\\"CommuteImage.jpeg\\")",
                }
              }
              title="Commute"
            />
            <div
              className="MuiCardContent-root-227 InfoPage-cardContent-10"
            >
              <h2
                className="MuiTypography-root-62 MuiTypography-headline-67 MuiTypography-gutterBottom-80"
              >
                Enjoy Commuting
              </h2>
              <p
                className="MuiTypography-root-62 MuiTypography-body1-71"
              >
                Driving the same route on a daily basis is a tedious task but when you're driving with new friends it will become something you look forward to. Our Carpool Platform makes numerous provisions to ensure you only travel with people you feel comfortable around, so the only question that remains is - why drive alone when you can drive together?
              </p>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-item-92 MuiGrid-grid-sm-6-139 MuiGrid-grid-md-4-151 MuiGrid-grid-lg-3-164"
        >
          <div
            className="MuiPaper-root-22 MuiPaper-elevation1-25 MuiPaper-rounded-23 MuiCard-root-224 InfoPage-card-8"
          >
            <div
              className="MuiCardMedia-root-225 InfoPage-cardMedia-9"
              src={undefined}
              style={
                Object {
                  "backgroundImage": "url(\\"RoadImage.jpeg\\")",
                }
              }
              title="Image title"
            />
            <div
              className="MuiCardContent-root-227 InfoPage-cardContent-10"
            >
              <h2
                className="MuiTypography-root-62 MuiTypography-headline-67 MuiTypography-gutterBottom-80"
              >
                Clean our Roads
              </h2>
              <p
                className="MuiTypography-root-62 MuiTypography-body1-71"
              >
                If you've ever had to sit in peak hour traffic you'll know what a pain it can be. A big reason for unnecessary traffic is that there are too many cars with one driver and a host of empty seats. By driving together, we are reducing the number cars on our roads, not only minimizing traffic, but minimizing the environmental impact our vehicles have.
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>,
  <footer
    className="InfoPage-footer-11"
  >
    <h2
      className="MuiTypography-root-62 MuiTypography-title-68 MuiTypography-gutterBottom-80 MuiTypography-alignCenter-76"
    >
      Created By
    </h2>
    <p
      className="MuiTypography-root-62 MuiTypography-subheading-69 MuiTypography-colorTextSecondary-86 MuiTypography-alignCenter-76"
    >
      The Brogrammers Development Team
    </p>
  </footer>,
]
`;
